libtransmission/peer-mgr.cc

```diff
@@ -2254,7 +2254,7 @@ void rechokeUploads(tr_swarm* s, uint64_t const now)
 
         if (auto const n = std::size(rand_pool); n != 0)
         {
-            auto* c = rand_pool[tr_rand_int_weak(n)];
+            auto* c = rand_pool[salter() % n];
```

Comment:
For the sake of avoiding future unknown and unseen issues, I am fundamentally against the usage of `%` when dealing with random values. Let's just use `arc4random_uniform(n)` when it's available.

You'd need to have a demonstration of a **noticeable** performance gain to ever adopt `%`.